vignettes/themes.Rmd

```diff
@@ -10,159 +10,83 @@ vignette: >
-demo_palette(protan(vlaanderen_palette()))
+```{r eval = FALSE, echo = TRUE}
+library(INBOtheme)
 ```
 
-```{r vlaanderen-palette-tritan, fig.cap = "Simulation of tritan model of colorblindness."}
-demo_palette(tritan(vlaanderen_palette()))
-```
+In case you need to change the default font size or background add the code below after loading the package.
 
-```{r vlaanderen-palette-gray, fig.cap = "Desaturated colours."}
-demo_palette(desaturate(vlaanderen_palette()))
+```{r eval = FALSE, echo = TRUE}
+theme_set(theme_inbo(base_size = 12, transparent = FALSE))
 ```
 
-### `nara_palette`
 
-```{r nara-palette-deutan, fig.cap = "Simulation of deutan model of colorblindness."}
-demo_palette(deutan(nara_palette()))
-```
+### Using a non-default theme from `INBOtheme`
 
-```{r nara-palette-protan, fig.cap = "Simulation of protan model of colorblindness."}
-demo_palette(protan(nara_palette()))
-```
+Add the following code after loading the `ggplot2` or `tidyverse` package and before creating the `ggplot2` plots.
 
-```{r nara-palette-tritan, fig.cap = "Simulation of tritan model of colorblindness."}
-demo_palette(tritan(nara_palette()))
+```{r eval = FALSE, echo = TRUE}
+library(INBOtheme)
+theme_set(theme_vlaanderen2015(base_size = 12, transparent = FALSE))
+switch_colour(vl_darkyellow)
 ```
 
-```{r nara-palette-gray, fig.cap = "Desaturated colours."}
-demo_palette(desaturate(nara_palette()))
+```{r eval = FALSE, echo = TRUE}
+library(INBOtheme)
+theme_set(theme_elsevier(base_size = 12, transparent = FALSE))
+switch_colour("black")
 ```
 
-### `traffic_palette`
+# Prepare the code and data
 
-```{r traffic-palette-deutan, fig.cap = "Simulation of deutan model of colorblindness."}
-demo_palette(deutan(traffic_palette(10)))
-```
-
-```{r traffic-palette-protan, fig.cap = "Simulation of protan model of colorblindness."}
-demo_palette(protan(traffic_palette(10)))
+```{r knitr, include = FALSE}
+library(knitr)
+opts_chunk$set(
+  echo = FALSE, message = FALSE, warning = FALSE
+)
 ```
 
-```{r traffic-palette-tritan, fig.cap = "Simulation of tritan model of colorblindness."}
-demo_palette(tritan(traffic_palette(10)))
+```{r initialise, echo = TRUE}
+library(ggplot2)
+library(INBOtheme)
 ```
 
-```{r traffic-palette-gray, fig.cap = "Desaturated colours."}
-demo_palette(desaturate(traffic_palette(10)))
+```{r prepare-data, echo = TRUE}
+# Prepare the data for the figures
+d <- data.frame(
+  x = "Sphinx of black quartz, judge my vow",
+  y = "Sphinx of\nblack quartz,\njudge my vow"
+)
+data("esoph")
+cancer <- esoph
+cancer$Age <- cancer$agegp
+cancer$Alcohol <- cancer$alcgp
+cancer$Tabacco <- cancer$tobgp
+cancer$Proportion <- cancer$ncases / (cancer$ncases + cancer$ncontrols)
 ```
 
-# Available themes
+# Default: `theme_inbo()`
 
-## Default: `theme_inbo`
+## `theme_inbo()` with default background
 
 ```{r set_theme_inbo, echo = TRUE}
-theme_set(theme_inbo())
+theme_set(theme_inbo(7))
```

Comment:
```suggestion
theme_set(theme_inbo(base_size = 7))
```
It is unclear for a reader why this `7` is there.  As suggested, just adding the argument already solves my concern.
On the other hand, as you seem to be using the 7 everywhere, it seems this is a perfect size for a default, or am I mistaken here?

Reply:
A good `base_size` depends on the output format. See https://github.com/inbo/INBOmd/blob/33fdf6c06ec1f357ba048c5f24831726be2ea3b1/inst/rmarkdown/templates/report/skeleton/skeleton.Rmd#L118 for a few examples. The current default works best for interactive use.

Reply:
Ok, in this case just consider adding `base_size =` to make clear for the user where the `7` stands for.  (I checked it in the function documentation, but as a reader I would rather like it to be clear in the vignette itself.)